config/routes.rb

```diff
@@ -0,0 +1,16 @@
+Rails.application.routes.draw do
+  mount Rswag::Ui::Engine => '/api-docs'
+  mount Rswag::Api::Engine => '/api-docs'
+  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  namespace :api, defaults: { format: :json } do
+    namespace :v1 do
+      resources :items, only: [:index, :show, :create, :destroy]
+      resources :users, only: [:create]
+
+      get '/:username/reservations', to: 'reservations#index'
```

Comment:
A better way to name this route (endpoint), in fact the more proper way, is the following:

```
resources :users, param: :username, only [:create] do
  resources :reservations, only: [:index, :create, :destroy]
end
```

You can then just use the exact same "/<username>/reservations" URL (but different verbs) for both index and create (GET and POST) while you can use "/<username>/reservations/<id>" for destroy.